ecs-agent/gmsacredclient/credentialsfetcherclient.go

```diff
@@ -62,6 +62,116 @@ type CredentialsFetcherResponse struct {
+// to check the health status of daemon
+func (c CredentialsFetcherClient) HealthCheck(ctx context.Context, serviceName string) (string, error) {
+	if len(serviceName) == 0 {
+		return "service name should not be empty", status.Errorf(codes.InvalidArgument, "service name should not be empty")
+	}
+	defer c.conn.Close()
+	client := pb.NewCredentialsFetcherServiceClient(c.conn)
+
+	request := &pb.HealthCheckRequest{Service: serviceName}
+
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
+	defer cancel()
+
+	response, err := client.HealthCheck(ctx, request)
+	if err != nil {
+		seelog.Errorf("credentials-fetcher daemon is inactive")
+		return "", err
+	}
+	seelog.Infof("credentials-fetcher daemon is active")
+
+	return response.GetStatus(), nil
+}
+
+// AddKerberosArnLease() invokes credentials fetcher daemon running on the host
+// to create kerberos tickets associated with gMSA accounts
+func (c CredentialsFetcherClient) AddKerberosArnLease(ctx context.Context, credentialspecsArns []string, accessKeyId string, secretKey string, sessionToken string, region string) (CredentialsFetcherArnResponse, error) {
+	if len(credentialspecsArns) == 0 {
+		return CredentialsFetcherArnResponse{}, status.Errorf(codes.InvalidArgument, "credentialspecArns should not be empty")
+	}
+
+	if len(accessKeyId) == 0 || len(secretKey) == 0 || len(sessionToken) == 0 || len(region) == 0 {
+		return CredentialsFetcherArnResponse{}, status.Errorf(codes.InvalidArgument, "accessid, secretkey, sessiontoken or region should not be empty")
+	}
+
+	defer c.conn.Close()
+	client := pb.NewCredentialsFetcherServiceClient(c.conn)
+
+	request := &pb.KerberosArnLeaseRequest{CredspecArns: credentialspecsArns, AccessKeyId: accessKeyId, SecretAccessKey: secretKey, SessionToken: sessionToken, Region: region}
+
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(c.timeout))
+	defer cancel()
+
+	response, err := client.AddKerberosArnLease(ctx, request)
+	if err != nil {
+		seelog.Errorf("could not create kerberos tickets: %v", err)
+		return CredentialsFetcherArnResponse{}, err
+	}
+	seelog.Infof("created kerberos tickets and associated with LeaseID: %s", response.GetLeaseId())
+
+	credentialsFetcherResponse := CredentialsFetcherArnResponse{
+		LeaseID:            response.GetLeaseId(),
+		KerberosTicketsMap: make(map[string]string),
+	}
+
+	for _, value := range response.GetKrbTicketResponseMap() {
+		credSpecArn := value.GetCredspecArns()
+		_, ok := credentialsFetcherResponse.KerberosTicketsMap[credSpecArn]
+		if ok {
+			seelog.Infof("credentialspec arn already added to map: %s", credSpecArn)
+		} else {
+			credentialsFetcherResponse.KerberosTicketsMap[credSpecArn] = value.GetCreatedKerberosFilePaths()
+		}
+	}
+
+	return credentialsFetcherResponse, nil
+}
+
+// RenewKerberosArnLease() invokes credentials fetcher daemon running on the host
+// to create kerberos tickets associated with gMSA accounts
```

Comment:
The description of this method is the same as that of `AddKerberosArnLease` method. Can we please update the descriptions so that the difference between the two methods is clear.